ui/apps/platform/src/services/ComplianceScanConfigurationService.ts

```diff
@@ -239,3 +240,50 @@ export function listComplianceScanConfigClusterProfiles(
         )
         .then((response) => response.data);
 }
+
+/*
+ *  Fetches a list of snapshots (scan executions) from a given scan config
+ */
+
+export type FetchComplianceReportHistoryServiceProps = {
+    id: string;
+    query: string;
+    page: number;
+    perPage: number;
+    sortOption: ApiSortOption;
+    showMyHistory: boolean;
+};
+
+export type ReportHistoryResponse = {
+    complianceReportSnapshots: ComplianceReportSnapshot[];
+};
+
+export function fetchComplianceReportHistory({
+    id,
+    query,
+    page,
+    perPage,
+    sortOption,
+    showMyHistory,
+}: FetchComplianceReportHistoryServiceProps): Promise<ComplianceReportSnapshot[]> {
+    const params = qs.stringify(
+        {
+            reportParamQuery: {
+                query,
+                pagination: getPaginationParams({ page, perPage, sortOption }),
+            },
+        },
+        { arrayFormat: 'repeat', allowDots: true }
+    );
```

Comment:
Mostly because of how the backend is set up right now. I am unaware of the specifics but If we're utilizing the same pattern as VM Reports, we'll have to keep it this way for consistency. 

It seems like we do have a few APIs that have a specific key with the `query` and `pagination` values under it. `v2/compliance/scan/results/profiles/:profileId/clusters/:clusterId` comes to mind. That one seems even weirder. It does `query.query` and `query.pagination`. This is a good discussion topic to go over with Backend. Maybe we can talk about it for the NA F2F meeting coming up. Keeping APIs and query params consistent would be a big help to the way UI and Backend interact in future feature development.